### PR TITLE
#162414275 Allow a User to Comment on Article

### PR DESCRIPTION
**Pull request by tejiri4**
#### What does this PR do?
This PR allows a user to comment on an article
#### Description of Task to be completed?

- Create an authenticated route to add a comment
- Create comment controller
- Write test
#### How should this be manually tested?
- clone the repo
- cd into it
- npm install
- npm run test
#### What are the relevant pivotal tracker stories?
[create comment](https://www.pivotaltracker.com/story/show/162414275)

### CODE REVIEW

File: server/controllers/CommentController.js [hunk 1]
```diff
@@ -28,7 +33,15 @@ class CommentController {
...
+      }
+      await Comment.create({
+        id,
+        content,
+        userId,
+        articleId
+      });
       const getComment = await Comment.findOne({
         where: {
           articleId,
```
Comment (ebzeal) [inline server/controllers/CommentController.js]: Nice work TJ! 
I feel you should not need to getComment from the DB in order to confirm comment creation. You could just do
`const commentCreate = await Comment.create({
        id,
        content,
        userId,
        articleId
      });`
and check 
`if(commentCreate)`

File: server/controllers/CommentController.js [hunk 2]
```diff
@@ -1,25 +1,30 @@
...
+   * @static
+   * @param {*} req Express Request object
+   * @param {*} res Express Response object
+   * @returns {object} Json response
+   * @memberof CommentController
+   */
   static async createComment(req, res) {
     try {
-      const { userId, articleId, content } = req.body;
+      const token = req.headers.authorization.split(' ')[1];
```
Comment (ebzeal) [inline server/controllers/CommentController.js:23]: Why do you split Authorization? Are we using Bearer or a prefix before our token?

File: server/controllers/CommentController.js [hunk 3]
```diff
@@ -39,7 +52,7 @@ class CommentController {
         return response(res, 201, 'success', 'Comment created', null, getComment.dataValues);
```
Comment (ebzeal) [inline server/controllers/CommentController.js]: Nice work TJ!
I feel you should not need to getComment from the DB in order to confirm comment creation. You could just do
const commentCreate = await Comment.create({ id, content, userId, articleId });
and check
if(commentCreate)

Comment (tejiri4) [reply server/controllers/CommentController.js]: Yes comment created are gotten from the response body with getComment.datavalues

File: server/controllers/CommentController.js (hunk 2 above)
Comment (tejiri4) [reply server/controllers/CommentController.js:23]: Yes bearer comes with the token

File: server/controllers/CommentController.js (hunk 3 above)
Comment (tejiri4) [reply server/controllers/CommentController.js]: screenshot 2019-01-18 at 5 47 38 am

File: .eslintignore [hunk 4]
```diff
@@ -1,5 +1,4 @@
-
 build/
 server/migrations
 server/seeders
-server/test
+server/test/*
```
Comment (rafmme) [inline .eslintignore]: No new line here

File: .gitignore [hunk 5]
```diff
@@ -49,3 +49,8 @@ package-lock.json
 
 # personal files
 swagger.1.yaml
+
+#code formatter
+.prettierrc
+
+.vscode
```
Comment (rafmme) [inline .gitignore]: no new line

**Review by rafmme — COMMENTED** [review]
LGTM, yet

**Comment by Bobsar0** [comment]
LGTM

File: server/controllers/CommentController.js (hunk 2 above)
Comment (ebzeal) [reply server/controllers/CommentController.js:23]: Our tokens are not returning Bearer

File: server/controllers/CommentController.js (hunk 3 above)
Comment (ebzeal) [reply server/controllers/CommentController.js]: Great, Thanks!

**Review by ebzeal — COMMENTED** [review]
LGTM

File: server/middlewares/AuthMiddleware.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
+import TokenManager from '../helpers/TokenManager';
+import response from '../helpers/response';
```
[the remark this answers is not in the document]
Comment (tejiri4) [reply server/middlewares/AuthMiddleware.js:2]: resolved

**Pull request closed without merging**